tests/index.spec.js

```diff
@@ -31,6 +31,14 @@ describe('Store Mock', () => {
     expect(store.state.nested).toEqual(state.nested)
   })
 
+  it('deeply copies the getters', () => {
+    const getters = { nested: { n: 0 } }
```

Comment:
When do you have such `getters`?

Reply:
Nested getters are not namespaces like `state` is and with modules, the name convention is different: https://github.com/posva/vuex-mock-store#getters

Reply:
Maps are faster than arrays (and keep code clean), that's my use case. They work fine through vuex and this library is helpful for changing getters. I am not using nested modules, please check both unit tests

Reply:
I read both unit tests but I can't picture what Vuex use case these are trying to mock. Please provide a boiled down reproduction so I can understand what bug you are trying to fix

Reply:
Im making a small project that acts as a small bookie offering odds on coin flips. I have an object where a key is a game ID, and the value is another object with a market array and game state etc. There is a websocket connection populating these for multiple games

Reply:
I see, but why would you write

```js
store.getters.positions.x = 3
```

If you want to mock the getter, you should mock the whole value returned by the getter:

```js
store.getters.position = { x: 3 }
```

Reply:
I am mocking the whole value, but I don't know X is when i initialise the object.
In the case of a game id it can be any positive number. so initially the getter is {}, after the first websocket messages it might be:
```
{
  10001: {
     Markets: [],
     Status: 'Open',
  }
}
```
few seconds later there's going to be another game id in there 10002 etc.
I know this behaviour is not allowed in TS, which is why there's a `// @ts-ignore`

Reply:
If in your real code you are setting the value returned by a getter, yes, that's not supposed to be done, so I think having to manually replace the whole value seems like a good thing

Reply:
This is what I am doing in the code:
![image](https://user-images.githubusercontent.com/9282777/79576864-d04a6e80-80bb-11ea-90f8-def81d66f15f.png)
You can see first the object has game id 88, then both 7 and 88
So I'm modifying the state, and the reason this library looks like a good choice is because of I don't have to change the state and can go for the helpers directly :)

from README.md
> getters
> Store getters. You can directly modify it to change a value:
> 
> store.getters.upperCaseName = 'JEFF'

Reply:
I don't understand: you are talking about the `state` now but it's handled completely different.

My point is: there is no need to deep clone the `getters` because they are not nested (like state). So I'm trying to see if there is a case I forgot to handle.

> from README.md

Exactly, that's why one should do `store.getters.position = { x: 3 }`

Reply:
Ok, I get your point. State is indeed nested, but if the state is only used by the getters in the original implementation (as opposite to a test file), then they should be nested too?

See below code for the test I'm trying to run. Even though reset is being called beforeEach test, it's still throwing.

![image](https://user-images.githubusercontent.com/9282777/79584067-c9285e00-80c5-11ea-89dc-e21f188976fe.png)

> state.js
```js
export default {
    games: {},
    isConnected: false,
    connectionProgress: 0,
    latestGameId: 0,
}
```
> getters.js
```js
export default {
    allGames(state) {
        return state.games
    },
    latestGameId(state) {
        return state.latestGameId
    },
    currentGame(state) {
        return state.games[state.latestGameId]
    },
    isConnected(state) {
        return state.isConnected
    },
    connectionProgress(state) {
        return state.connectionProgress
    },
}
```
As you can see I never use the `mapState` helper in my vue files, the state is only used in the getters and mutations

Reply:
But, you shouldn't do

```js
store.getters.allGames[100] = 'something'
```

because getters are read only in regular code, instead you should do

```js
store.getters.allGames = { ...store.getters.allGames, 1000: 'something' }
```

if you want to change a getter

Reply:
Don't wanna sound like a broken record here. I understand Vue getters don't have setters which is exactly my use for this library. It's what it promises in the readme.md. The code snippet from you does work, but it's merely a hack my side that replaces what should be replaced by the library's reset method. I've replaced my code with `...` now but I am hoping this PR fixes what I think the library promises to do. Thanks

Reply:
I appreciate the pr but I think the current design prevents
an anti pattern while still being able to get around it easily